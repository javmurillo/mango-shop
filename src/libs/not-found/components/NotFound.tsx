import { Alert, Navbar } from 'react-bootstrap';

export const NotFound = (): JSX.Element => {
  return (
    <div aria-label="not found">
      <Navbar />
      <Alert variant="dark" aria-label="not found alert">
        The page you are looking for was not found!
      </Alert>
    </div>
  );
};
