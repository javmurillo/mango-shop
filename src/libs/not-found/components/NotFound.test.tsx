import { cleanup, render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import { NotFound } from './NotFound';

afterEach(cleanup);

const setup = () => {
  const utils = render(<NotFound aria-label="not found" />);
  const navbar = utils.getByLabelText('not found');
  return {
    navbar,
    ...utils,
  };
};

describe('<NotFound/> tests', () => {
  test('renders correctly', () => {
    const { navbar } = setup();
    expect(navbar).toBeInTheDocument();
  });

  test('shoud match snapshot', () => {
    const testRendererJson = TestRenderer.create(
      <NotFound aria-label="not found" />
    ).toJSON();
    expect(testRendererJson).toMatchSnapshot();
  });

  test('not found alert is present', () => {
    setup();
    const mangoLogo = screen.getByLabelText('not found alert');
    expect(mangoLogo).toBeInTheDocument();
  });
});
