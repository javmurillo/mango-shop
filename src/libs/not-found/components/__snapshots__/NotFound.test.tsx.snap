// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> tests shoud match snapshot 1`] = `
<div
  aria-label="not found"
>
  <nav
    className="navbar navbar-expand navbar-light"
  />
  <div
    aria-label="not found alert"
    className="fade alert alert-dark show"
    role="alert"
  >
    The page you are looking for was not found!
  </div>
</div>
`;
